pkg/apis/sources/v1beta1/ping_validation.go

```diff
@@ -46,6 +49,20 @@ func (cs *PingSourceSpec) Validate(ctx context.Context) *apis.FieldError {
 		}
 	}
 
+	pingConfig := config.FromContextOrDefaults(ctx)
+
+	pingDefaults := pingConfig.PingDefaults.GetPingConfig()
+
+	if bsize := len(cs.JsonData); bsize > pingDefaults.JsonDataMaxBytes {
+		fe := apis.ErrInvalidValue(fmt.Sprintf("the jsonData length of %d bytes exceeds limit set at %d.", bsize, pingDefaults.JsonDataMaxBytes), "jsonData")
+		errs = errs.Also(fe)
+	}
+
+	if len(cs.JsonData) > 0 && !json.Valid([]byte(cs.JsonData)) {
```

Comment:
this is not backward compatible. jsonData is a misnomer

Reply:
We discussed in slack conversation, you agreed to json validation. Are you now rethinking that?

Reply:
./kn   source ping  create --help 
...
-d, --data string               **Json** data to send

does not say any text string.

Reply:
well that's `kn`, not what the implementation is currently supporting. Let's get our story straight for v1